dandiapi/api/views/asset.py

```diff
@@ -527,13 +531,13 @@ def list(self, request, *args, **kwargs):
             glob_pattern = re.escape(glob_pattern)
             queryset = queryset.filter(path__iregex=glob_pattern.replace('\\*', '.*'))
 
+        # Paginate if necessary
         page = self.paginate_queryset(queryset)
         if page is not None:
-            serializer = self.get_serializer(page, many=True)
-            return self.get_paginated_response(serializer.data)
+            queryset = page
```

Comment:
I know nothing about django/querysets and serializers, but 
- when page could be `None`?
- is it expected to reuse original `queryset` from above if page is None?

logically to me page would be `None` if there is no assets, but then why do we need any serializers and original querysets at all then?

Reply:
I'm not sure under what circumstances `page` is None, but it's seemingly possible, since this code is taken almost verbatim from the source code for the Django Rest Framework [`ListModelMixin`](https://github.com/encode/django-rest-framework/blob/a1b35bb44b7c9251c8b7bc995aa6598044f1d3ef/rest_framework/mixins.py#L33-L46) class.

Reply:
Referenced code is
```python

        page = self.paginate_queryset(queryset)
        if page is not None:
            serializer = self.get_serializer(page, many=True)
            return self.get_paginated_response(serializer.data)

        serializer = self.get_serializer(queryset, many=True)
        return Response(serializer.data)
```
and it
- doesn't overload queryset (readability)
- returns response in case of page being None 

I guess should be matched here?